### PR TITLE
feat(Experiments): Allow users to edit and preview no-code experiments from experiment page.

### PR DESCRIPTION
**Pull request by Phanatic**
## Problem

In the past, creating a no-code experiment required a user to navigate to the `Toolbar` page and then launch the toolbar from there to edit an existing experiment. 

This PR instead adds a custom `implementation` section to the experiment page that allows a user to bring up the toolbar with the experiment pre-selected. It also adds a `preview variant` button to each variant of the web experiment.

### 1. Editing a web experiment from its individual page.

On the edit page of a web experiment, this is what you see now for `implementation`

image

Clicking on `Edit Web experiment on website` brings up this dialog that shows the authorized URLs for the posthog project

image

Selecting a URL takes one to the domain with the experiment pre-selected

image

### 2. Previewing web experiment variants from an experiment page.
The Distribution table now has an additional column if its a web experiment, all widths adjusted accordingly.

image

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Comment by jurajmajerik** [comment]
Not sure if this is still on a todo list, but transforms for a variant still isn't applied correctly here:

https://github.com/user-attachments/assets/71d71115-0b7d-4e7f-a596-cffbaeab179f

**Comment by jurajmajerik** [comment]
When saving a new experiment, an empty error banner flickers before successfully saving:

https://github.com/user-attachments/assets/28ff273a-eac2-4626-aa26-c323cd5dd8f7

File: frontend/src/scenes/experiments/WebExperimentImplementationDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                        <LemonButton
+                            size="small"
+                            type="secondary"
+                            onClick={onSelectElement}
+                            sideIcon={<IconOpenInApp />}
+                        >
+                            Edit Web experiment on website
+                        </LemonButton>
+                    </div>
+                </div>
```
Comment (jurajmajerik) [inline frontend/src/scenes/experiments/WebExperimentImplementationDetails.tsx:49]: Suggesting a bit more specific copy + small alignments:

image

```suggestion
                <div className="border p-6 rounded bg-bg-light space-y-2">
                    <div className="font-semibold leading-tight text-base text-current">
                        Define variant changes directly on your website
                    </div>
                    <div className="text-sm text-muted">
                        Use our toolbar to select elements and apply transformations for each variant.
                    </div>
                    <div>
                        <LemonButton
                            size="small"
                            type="secondary"
                            onClick={onSelectElement}
                            sideIcon={<IconOpenInApp />}
                        >
                            Launch toolbar on your website
                        </LemonButton>
                    </div>
                </div>
```

**Comment by jurajmajerik** [comment]
Working well, some comments :)

**Comment by Phanatic** [comment]
@jurajmajerik thanks for taking a look!

re : https://github.com/PostHog/posthog/pull/25877#issuecomment-2444731147

The transform the HTML is overriding the transform for the text. Maybe I should order it so that the most significant changes happen first and then the lower priority. As an example HTML Changes -> CSS changes -> Text changes. 

This way, the Text changes are always applied.

**Comment by Phanatic** [comment]
Good catch on https://github.com/PostHog/posthog/pull/25877#issuecomment-2444738825 
I just fixed this, it was an issue with checking if any errors existed on the logic before saving. 
no_flashing_error

File: frontend/src/scenes/experiments/WebExperimentImplementationDetails.tsx (hunk 1 above)
Comment (Phanatic) [reply frontend/src/scenes/experiments/WebExperimentImplementationDetails.tsx]: applied changes!

**Review by jurajmajerik — APPROVED** [review]

**Pull request merged**